osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/PushSetupAlarmConverter.java

```diff
@@ -65,11 +96,38 @@ public PushSetupAlarm convertFrom(
     }
     final PushSetupAlarm.Builder builder = new PushSetupAlarm.Builder();
 
-    final String destination = source.getHost() + ":" + source.getPort();
-    final SendDestinationAndMethod sendDestinationAndMethod =
-        new SendDestinationAndMethod(
-            TransportServiceType.TCP, destination, MessageType.MANUFACTURER_SPECIFIC);
-    builder.withSendDestinationAndMethod(sendDestinationAndMethod);
+    if (source.getHost() != null && source.getPort() != null) {
+      final String destination = source.getHost() + ":" + source.getPort();
+      final SendDestinationAndMethod sendDestinationAndMethod =
+          new SendDestinationAndMethod(
+              TransportServiceType.TCP, destination, MessageType.MANUFACTURER_SPECIFIC);
+      builder.withSendDestinationAndMethod(sendDestinationAndMethod);
+    }
+
+    final List<PushObject> sourcePushObjects = source.getPushObjectList();
```

Comment:
You already retrieved the pushobjectlist from source and added it to pushSetupAlarm

Reply:
That's in the other function.